### PR TITLE
Compute of Extension name can throw error

### LINKED ISSUE #92: Compute of Extension name can throw error
JDT LS extension computes the extension name from the JAR name. There are some error with this compute which caues the problem that Quarkus properties is ignored.

See error trace https://travis-ci.org/redhat-developer/quarkus-ls/jobs/585973569#L402

### PR DESCRIPTION
**Pull request by angelozerr**
Compute of Extension name can throw error

Fixes #92

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
@fbricon @xorye this PR is very important, is there any chance that you could review it before the release?

File: quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java [hunk 1]
```diff
@@ -211,8 +211,15 @@ private static void processConfigProperty(IJavaElement javaElement, IAnnotation
...
 				IType fieldClass = findType(field.getJavaProject(), fieldTypeName);
 				String defaultValue = getAnnotationMemberValue(configPropertyAnnotation, "defaultValue");
-				addField(field, fieldTypeName, fieldClass, propertyName, defaultValue, converter, javadocCache, null,
-						quarkusProperties, monitor);
+				// Location (JAR, src)
+				IPackageFragmentRoot packageRoot = (IPackageFragmentRoot) javaElement
+						.getAncestor(IJavaElement.PACKAGE_FRAGMENT_ROOT);
+				String location = packageRoot.getPath().toString();
+				// Quarkus Extension name
+				String extensionName = JDTQuarkusUtils.getExtensionName(location);
```
Comment (fbricon) [inline quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: might as well provide JDTQuarkusUtils.getExtensionName(javaElement)

File: quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java [hunk 2]
```diff
@@ -211,8 +211,15 @@ private static void processConfigProperty(IJavaElement javaElement, IAnnotation
 				String fieldTypeName = getResolvedTypeName(field);
 				IType fieldClass = findType(field.getJavaProject(), fieldTypeName);
 				String defaultValue = getAnnotationMemberValue(configPropertyAnnotation, "defaultValue");
-				addField(field, fieldTypeName, fieldClass, propertyName, defaultValue, converter, javadocCache, null,
-						quarkusProperties, monitor);
+				// Location (JAR, src)
+				IPackageFragmentRoot packageRoot = (IPackageFragmentRoot) javaElement
```
Comment (fbricon) [inline quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: can packageroot be null?

File: quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java (hunk 1 above)
Comment (angelozerr) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: I need location & extensionName both.

File: quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java (hunk 2 above)
Comment (angelozerr) [reply quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java]: Never seen the case and all code that I see doesn't check the NPE

**Pull request merged**